### PR TITLE
dada2 fixes

### PR DESCRIPTION
**Pull request by bernt-matthias**
detected here https://github.com/galaxyproject/tools-iuc/issues/2701

- data manager: real bug fix
- bug fix in makeSequencTable: cheetah for testing `plot` variable was wrong
- allow larger delta for tests on Rdata and pdf (mostly 1/2 file size, but anyway less then the file size .. essentially now a test for file presence)

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: data_managers/data_manager_dada2/data_manager/data_manager.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def remote_dataset(dataset, outjson):
         data_manager_entry['path'] = dataset + ".species"
         data_manager_json["data_tables"]["dada2_species"] = data_manager_entry
 
-    with file(outjson, 'w') as jf:
+    with open(outjson, 'w') as jf:
```
Comment (mvdbeek) [inline data_managers/data_manager_dada2/data_manager/data_manager.py]: To make this more consistent for python versions that don't guarantee dictionary sort order (python < 3.6) I'd change the line below to `jf.write(json.dumps(data_manager_json, sort_keys=True))`

File: data_managers/data_manager_dada2/data_manager/dada2_fetcher.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<tool id="dada2_fetcher" name="dada2 data manager" tool_type="manage_data" version="0.0.7">
+<tool id="dada2_fetcher" name="dada2 data manager" tool_type="manage_data" version="0.0.8">
```
Comment (mvdbeek) [inline data_managers/data_manager_dada2/data_manager/dada2_fetcher.xml]: Can you add a python requirement (I add python 3.7 for the data managers I already worked on) here and bump the `profile` to 18.09, which should be the first one that will use containers for data managers ?

File: tools/dada2/dada2_makeSequenceTable.xml [hunk 3]
```diff
@@ -15,7 +15,7 @@
 @WRITE_FOO@
 
 library(dada2, quietly=T)
-#if $plot == "yes"
+#if $plot
```
Comment (mvdbeek) [inline tools/dada2/dada2_makeSequenceTable.xml]: That should have been fixed by https://github.com/galaxyproject/galaxy/pull/9052/ ... can you check if that is still necessary ? It's fine the way you're doing it here but better to be aware if we're breaking Galaxy tools in unforeseen ways.

File: data_managers/data_manager_dada2/data_manager/data_manager.py (hunk 1 above)
Comment (mvdbeek) [reply data_managers/data_manager_dada2/data_manager/data_manager.py]: (that may also require updating any test output data if the order changes)

File: data_managers/data_manager_dada2/data_manager/dada2_fetcher.xml (hunk 2 above)
Comment (bernt-matthias) [reply data_managers/data_manager_dada2/data_manager/dada2_fetcher.xml]: done

**Comment by bgruening** [comment]
@bernt-matthias can you maybe use https://github.com/galaxyproject/galaxy/pull/8852 ... and just check for size. This way we also remove the Rdata from the repo.

**Comment by bernt-matthias** [comment]
Not sure if file size is the same on all systems since the files can be compressed with different methods or even uncompressed. 

We could fix to uncompressed. Which would be faster, but require a bit more space. My guess would be that the file size might be the same on all systems.

File: data_managers/data_manager_dada2/data_manager/data_manager.py (hunk 1 above)
Comment (bernt-matthias) [reply data_managers/data_manager_dada2/data_manager/data_manager.py]: consider it done

File: tools/dada2/dada2_makeSequenceTable.xml (hunk 3 above)
Comment (bernt-matthias) [reply tools/dada2/dada2_makeSequenceTable.xml]: seems to work with `planemo t --galaxy_branch release_19.09`, but not with `planemo t`. I guess the commits are not yet in master?

Anyway, I will restore it.

**Comment by bgruening** [comment]
You have a delta in there as well: https://github.com/galaxyproject/galaxy/pull/8852/files#diff-4191170733b24a9e99ba70b64c8b8787R30

**Comment by mvdbeek** [comment]
Oh yeah, it's not in master yet, I'll do that. There's also https://github.com/galaxyproject/galaxy/pull/9054 to look out for.

**Comment by mvdbeek** [comment]
master is updated now

**Comment by bernt-matthias** [comment]
@bgruening sounds good. just was wondering if --update_test_data updates the numbers ..? Probably not... :)

**Comment by bernt-matthias** [comment]
Seems that also https://github.com/galaxyproject/galaxy/pull/8852 has not found its way to master and 19.09 which would be necessary.

**Comment by bernt-matthias** [comment]
Anyway, I just updated the test and do not see the advantage here: We can't remove the test outputs, since subsequent steps of the dada2 pipeline use them as input. 

My last commit could be easily undone if you agree.

**Comment by bgruening** [comment]
Imho it makes a lot of sense for PDF ... that tools read Rdata is not good, security wise. But I guess this is unrelaed to this PR.

**Comment by bernt-matthias** [comment]
> that tools read Rdata is not good, security wise. But I guess this is unrelaed to this PR.

still did not get this point. as far as I see it nothing is executed on import.

**Comment by bernt-matthias** [comment]
@mvdbeek to make this work we need https://github.com/galaxyproject/galaxy/pull/8852 in master and 19.09 .. and I think also the xsd of planemo needs an update

**Comment by mvdbeek** [comment]
It isn't a bug, I don't think we can do a backport. But it might not be required if we start using using the proper galaxy packages instead of galaxy-lib in planemo. @nsoranzo was working on this.
Can we just work with the delta for now ?

**Comment by bernt-matthias** [comment]
yep. will reset to https://github.com/galaxyproject/tools-iuc/pull/2705/commits/770ea3840e37f02963dd3ca480846e6445f5bfc6

**Pull request merged**